user-guide/Advanced_Modules/DOM/DomHelper_class.md

```diff
@@ -54,6 +57,44 @@ if (!traceData.HasSucceeded())
 }
 ```
 
+### Multiple instances
+
+When multiple `DomInstances` need to get created, updated, or deleted, we recommend calling the *CreateOrUpdate* or *Delete* methods on a `DomInstance` CRUD helper component with a list of those `DomInstances`. This feature is available from DataMiner 10.4.2/10.5.0 onwards<!-- RN 37891 -->.
+
+For example, to create or update multiple `DomInstances`:
+
+```csharp
+// Create or update a list of DomInstances
+var createResult = helper.DomInstances.CreateOrUpdate(domInstances);
+```
+
+*CreateOrUpdate* will consider `DomInstances` that are stored already as `DomInstances` that need an update. The remaining `DomInstances` will be created. A mix of both is allowed.
+
+> [!TIP]
+> For more information and best practices for using these calls, see [Processing multiple DomInstances — examples](xref:DOM_BulkProcessing_Example).
+
+> [!IMPORTANT]
+> When designing the object model, consider if a high number of `DomInstances` might need to be processed quickly or need to be provisioned. If this is the case, we recommend avoiding related actions such as [launching script actions](xref:ExecuteScriptOnDomInstanceActionSettings) and [history tracking](xref:DOM_history).
+>
+> The number of `DomInstances` that can be passed to these methods [is limited to 100](#maximum-number-of-instances). Since those related actions might outlive these CRUD calls, keep in mind that repeating these operations in succession can still impact the stability of the system.
+
+#### Call result
+
+When the operation fails for one of the `DomInstances`, the result of those calls will contain the necessary information.
+
+- When `CreateOrUpdate` is called, the result will contain a list of `DomInstances` that were successfully created or updated. The trace data is available per `DomInstance` ID.
+- When `Delete` is called with a list of `DomInstances`, the result will contain a list of `DomInstance` IDs that were successfully deleted. The trace data is also available per `DomInstance` ID.
+
+If an issue occurs when an item gets created, updated, or deleted (e.g. validation), no exception will be thrown (even when `ThrowExceptionsOnErrorData` is *true*). The result of the call can be used to check for which `DomInstances` the call succeeded or failed. In addition, the trace data of that call is available and will contain the trace data for all processed `DomInstances`.
+
+In case the entire operation fails (e.g. when the storage layer fails while storing the data) a `CrudFailedException` is thrown. If, in that case, `ThrowExceptionsOnErrorData` was set to false, these calls will return `null` and the `TraceData` of the last call should be checked to get more details about the issue. For information on how to implement this flow, refer to the [Unexpected issue example](xref:DOM_BulkProcessing_Example#unexpected-issue).
+
+#### Maximum number of instances
```

Comment:
I'm a little confused by this section: the way I understand it, there is a limit of 100 instances, but there is also a limit defined in the MaxAmountBulkOperation property. How should I understand the difference between the two? Is the property used to define a stricter limit than the 100?

Reply:
The maximum number of DomInstances that can be passed to CreateOrUpdate and Delete is indeed 100. That same number is defined in MaxAmountBulkOperation. That property can't be changed, but that way it's easier to remember and to consult it.
Instead of hardcoding 100, that property can be used instead.